### PR TITLE
[Merged by Bors] - Add .list-layouts command

### LINKED ISSUE #18: Set default values for config options in configure
This would make them more discoverable with `snap get`, as suggested in #17.

### PR DESCRIPTION
**Pull request by wmww**
This implements setting default config values (fixes #18), generates a list of keyboard layouts during build and adds a new program that lists all the layouts along with their human-readable name. This removes the guesswork from selecting a layout if you don't have the codes memorized. The output looks like this:
```
wmww@wmwwXPS:~/code/snaps/ubuntu-frame-osk$ ubuntu-frame-osk.list-layouts 
ID            NAME

am            Armenian
am+phonetic   Armenian (phonetic)
ara           Arabic
be            Belgian
bg            Bulgarian
bg+phonetic   Bulgarian (traditional phonetic)
br            Portuguese (Brazil)
ch            German (Switzerland)
ch+de         (null)
ch+fr         French (Switzerland)
cz            Czech
cz+qwerty     Czech (QWERTY)
de            German
dk            Danish
epo           Esperanto
es            Spanish
es+cat        Catalan (Spain, with middle-dot L)
fi            Finnish
fr            French
gr            Greek
il            Hebrew
ir            Persian
it            Italian
it+fur        Friulian (Italy)
jp+kana       Japanese (Kana)
no            Norwegian
pl            Polish
ru            Russian
se            Swedish
th            Thai
ua            Ukrainian
us            English (US)
us+colemak    English (Colemak)
us+dvorak     English (Dvorak)

Run `snap set ubuntu-frame-osk layout=id1,id2,id3` to set layouts
```

### CODE REVIEW

File: list-layouts/main.c [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+        getenv("SNAP_NAME")
+    );
+}
+
+void g_log_ignore (
+    const gchar* log_domain,
+    GLogLevelFlags log_level,
+    const gchar* message,
+    gpointer user_data
+) {}
```
Comment (AlanGriffiths) [inline list-layouts/main.c:69]: Untidy:
```plain
../src/main.c:65:18: warning: unused parameter ‘log_domain’ [-Wunused-parameter]
   65 |     const gchar* log_domain,
      |     ~~~~~~~~~~~~~^~~~~~~~~~
../src/main.c:66:20: warning: unused parameter ‘log_level’ [-Wunused-parameter]
   66 |     GLogLevelFlags log_level,
      |     ~~~~~~~~~~~~~~~^~~~~~~~~
../src/main.c:67:18: warning: unused parameter ‘message’ [-Wunused-parameter]
   67 |     const gchar* message,
      |     ~~~~~~~~~~~~~^~~~~~~
../src/main.c:68:14: warning: unused parameter ‘user_data’ [-Wunused-parameter]
   68 |     gpointer user_data
      |     ~~~~~~~~~^~~~~~~~~
```

**Review by AlanGriffiths — APPROVED** [review]
There's no copyright in `list-layouts/main.c`, so from an abundance of caution, could you confirm it is GPL3 as per LICENSE? (A comment here is enough.)

bors delegate

**Comment by wmww** [comment]
This supports multiple layouts, which means it fixes #11

**Review by AlanGriffiths — APPROVED** [review]

**Pull request closed without merging**